src/image.tsx

```diff
@@ -113,16 +119,31 @@ const Image = React.forwardRef<HTMLImageElement, ImageProps>(
         const needsWrapper = withCssAspectRatio && !supportsAspectRatio;
 
         const img = (
+            // https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/309
+            // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
             <img
                 {...getPrefixedDataAttributes(dataAttributes)}
                 ref={ref}
                 src={src}
                 className={classes.image}
                 alt={alt}
+                onError={(event) => {
+                    setIsError(true);
+                    onError?.(event);
+                }}
+                onLoad={(event) => {
+                    setIsError(false);
+                    onLoad?.(event);
+                }}
                 {...(!needsWrapper ? {width, height} : {})}
             />
         );
 
+        if (isError) {
+            // to avoid showing the broken image icon
+            return null;
```

Comment:
hmmm... the problem is that you'll cause a layout shift if the image load fails

Reply:
same problem happens now if the image is broken